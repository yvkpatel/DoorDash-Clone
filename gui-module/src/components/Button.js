const Button = ({text, color, onClick}) => {
    return(
        <button className='button'
        style={{background:color}}
        data-testid="button"
        onClick={onClick}>
            {text}
        </button>
    )
}

export default Button