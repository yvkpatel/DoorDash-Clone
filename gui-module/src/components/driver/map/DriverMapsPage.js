import React from "react";
import Header from "../../Header";
import Map from "./DriverMap";
import { propTypes } from "react-bootstrap/Image";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Ratio from "react-bootstrap/Ratio";
import Button from "react-bootstrap/Button";
import Stack from "react-bootstrap/Stack";
import { Link } from "react-router-dom";

const onPick = () => {
  // setPicked(true);
  //fetch("/order/status/set/001/1");
};

const onDeliver = () => {
  // setDelivered(true);
  // fetch("/order/status/set/001/2");
};

const DriverMapsPage = (props) => {
  const [disable, setDisable] = React.useState(false);
  return (
    <div data-testid="driver-map">
      <Header />

      <Container className="driverButtons" style={{ padding: "4vh" }}>
        <Row className="pickButton">
          <Col>
            <Stack gap={5}>
              <Button
                variant="custom"
                size="lg"
                style={{ backgroundColor: "#2DB6F0", color: "#FFFFFF" }}
                onClick={() => setDisable(true)}
                disabled={disable}
              >
                Pick Order
              </Button>
              <Link
                to={
                  disable
                    ? { pathname: "../driver-dash", state: {} }
                    : { pathname: propTypes.route }
                }
              >
                <Button
                  variant="custom"
                  size="lg"
                  style={{ backgroundColor: "#2DB6F0", color: "#FFFFFF" }}
                  disabled={!disable}
                >
                  Deliver Order
                </Button>
              </Link>
            </Stack>
          </Col>
          <Col>
            <Map />
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default DriverMapsPage;
