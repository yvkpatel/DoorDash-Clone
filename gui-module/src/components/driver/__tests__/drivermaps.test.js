import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import React, { useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import ReactDOM from "react-dom";
import Button from "../../Button";
import DriverMapsPage from "../map/DriverMapsPage";
import DriverDash from "../dashboard/DriverDashPage";
import "@testing-library/jest-dom/extend-expect";

afterEach(() => {
  cleanup();
});

test("should render driver menu component", () => {
  render(
    <BrowserRouter>
      <DriverMapsPage />
    </BrowserRouter>
  );
  const DriverMapsPageElement = screen.getByTestId("driver-map");
  expect(DriverMapsPageElement).toBeDefined();
});

test("Pick Order button is enabled after maps page renders", () => {
  render(
    <BrowserRouter>
      <DriverMapsPage />
    </BrowserRouter>
  );

  expect(screen.getByRole("button", { name: "Pick Order" })).toBeEnabled();
});

test("Deliver Order button is disabled after maps page renders", () => {
  render(
    <BrowserRouter>
      <DriverMapsPage />
    </BrowserRouter>
  );

  expect(screen.getByRole("button", { name: "Deliver Order" })).toBeDisabled();
});

test("After Pick Order button is clicked, it is disabled and Deliver Order button is enabled", () => {
  render(
    <BrowserRouter>
      <DriverMapsPage />
    </BrowserRouter>
  );

  userEvent.click(screen.getByText("Pick Order"));
  expect(screen.getByRole("button", { name: "Pick Order" })).toBeDisabled();
  expect(screen.getByRole("button", { name: "Deliver Order" })).toBeEnabled();
});

test("After Deliver Order button is clicked, go back to driver dashboard", () => {
  render(
    <BrowserRouter>
      <DriverMapsPage />
    </BrowserRouter>
  );
  fireEvent.click(screen.getByText("Pick Order"));
  fireEvent.click(screen.getByText("Deliver Order"));
  const newURL = window.location.href;
  console.log(newURL);
  expect(newURL).toBe("http://localhost/driver-dash");
});

test("matches snapshot", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <DriverMapsPage />
    </BrowserRouter>
  );
  expect(tree).toMatchSnapshot();
});
