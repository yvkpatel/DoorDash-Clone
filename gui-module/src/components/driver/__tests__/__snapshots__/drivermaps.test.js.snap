// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="driver-map"
>
  <header
    className="header"
  >
    <img
      alt="YUMI Logo"
      className="logo"
      src="logo-header.png"
    />
    <a
      className="accSettings"
      href="https://developer.mozilla.org/en-US/docs/Web/CSS"
    >
      Account Settings
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "background": "#2DB6F0",
          }
        }
      >
        Logout
      </button>
    </a>
  </header>
  <div
    className="driverButtons container"
    style={
      Object {
        "padding": "4vh",
      }
    }
  >
    <div
      className="pickButton row"
    >
      <div
        className="col"
      >
        <div
          className="vstack gap-5"
        >
          <button
            className="btn btn-custom btn-lg"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#2DB6F0",
                "color": "#FFFFFF",
              }
            }
            type="button"
          >
            Pick Order
          </button>
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="btn btn-custom btn-lg"
              disabled={true}
              style={
                Object {
                  "backgroundColor": "#2DB6F0",
                  "color": "#FFFFFF",
                }
              }
              type="button"
            >
              Deliver Order
            </button>
          </a>
        </div>
      </div>
      <div
        className="col"
      >
        <div>
          <img
            alt="map"
            src="map_filler_image.jpeg"
            style={
              Object {
                "display": "inline-flex",
                "height": "70vh",
                "width": "75vw",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
